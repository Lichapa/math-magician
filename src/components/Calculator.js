import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '0',
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick = (e) => {
    this.setState((previousState) => calculate(previousState, e));
  };

  render() {
    const { total, next, operation } = this.state;
    const display = () => {
      if (next) {
        return next;
      }
      if (operation) {
        return operation;
      }
      if (total) {
        return total;
      }
      return 0;
    };
    return (
      <div className="main-container" id="main-container">
        <div>
          <input
            type="number"
            className="display-calc grey"
            id="display-Calc"
            readOnly
            value={display()}
          />
        </div>
        <div className="buttons">
          <button
            type="button"
            className="white AC"
            onClick={() => this.handleClick('AC')}
          >
            AC
          </button>
          <button
            type="button"
            className="white plus-minus"
            onClick={() => this.handleClick('+/-')}
          >
            +/-
          </button>
          <button
            type="button"
            className="white percent"
            onClick={() => this.handleClick('%')}
          >
            %
          </button>
          <button
            type="button"
            className="orange divide"
            onClick={() => this.handleClick('รท')}
          >
            รท
          </button>
          <button
            type="button"
            className="white number-btn"
            onClick={() => this.handleClick('7')}
          >
            7
          </button>
          <button
            type="button"
            className="white number-btn"
            onClick={() => this.handleClick('8')}
          >
            8
          </button>
          <button
            type="button"
            className="white number-btn"
            onClick={() => this.handleClick('9')}
          >
            9
          </button>
          <button
            type="button"
            className="orange multiply"
            onClick={() => this.handleClick('x')}
          >
            x
          </button>
          <button
            type="button"
            className="white number-btn"
            onClick={() => this.handleClick('4')}
          >
            4
          </button>
          <button
            type="button"
            className="white number-btn"
            onClick={() => this.handleClick('5')}
          >
            5
          </button>
          <button
            type="button"
            className="white number-btn"
            onClick={() => this.handleClick('6')}
          >
            6
          </button>
          <button
            type="button"
            className="orange subtract"
            onClick={() => this.handleClick('-')}
          >
            -
          </button>
          <button
            type="button"
            className="white number-btn"
            onClick={() => this.handleClick('1')}
          >
            1
          </button>
          <button
            type="button"
            className="white number-btn"
            onClick={() => this.handleClick('2')}
          >
            2
          </button>
          <button
            type="button"
            className="white number-btn"
            onClick={() => this.handleClick('3')}
          >
            3
          </button>
          <button
            type="button"
            className="orange addition"
            onClick={() => this.handleClick('+')}
          >
            +
          </button>
          <button
            type="button"
            className="zero"
            onClick={() => this.handleClick('0')}
          >
            0
          </button>
          <button
            type="button"
            className="white decimal"
            onClick={() => this.handleClick('.')}
          >
            .
          </button>
          <button
            type="button"
            className="orange equal"
            onClick={() => this.handleClick('=')}
          >
            =
          </button>
        </div>
      </div>
    );
  }
}

export default Calculator;
