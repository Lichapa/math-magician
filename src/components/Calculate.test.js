import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
// import { PropTypes } from 'prop-types';
import Calculator from './Calculator';

it('Renders calculator', () => {
  const result = renderer.create(<Calculator />).toJSON();
  expect(result).toMatchSnapshot();
});

test('calls onClick prop when clicked', () => {
  render(<Calculator />);
  fireEvent.click(screen.getByText(/AC/i));
  expect(screen.getByTestId('in')).toBeTruthy();
});

afterEach(cleanup);
