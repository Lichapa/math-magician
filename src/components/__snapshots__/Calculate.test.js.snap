// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders calculator 1`] = `
<div
  className="main-container flex row center"
>
  <div>
    <h2>
      Let's do some math
    </h2>
  </div>
  <div
    className="calculator"
    id="main-container"
  >
    <div>
      <input
        className="display-calc grey"
        data-testid="in"
        id="display-Calc"
        readOnly={true}
        type="number"
        value={0}
      />
    </div>
    <div
      className="buttons"
    >
      <button
        className="white AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="white plus-minus"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="white percent"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="orange divide"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="white number-btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="white number-btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="white number-btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="orange multiply"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="white number-btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="white number-btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="white number-btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="orange subtract"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="white number-btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="white number-btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="white number-btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="orange addition"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="white decimal"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="orange equal"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
